src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.util.Scanner;
+
+/**
+ * <h1>Duke IP Project</h1>
+ *
+ * <p>A simple task managing chatbot. Functionalities include saving tasks, completing tasks, and </p>
+ */
+public class Duke extends Application {
+
+    private Button sendButton;
+    private TextField userInputTextField;
+    private ScrollPane scrollPane;
+    private VBox dialogueContainer;
+    private Scene scene;
+
+    /**
+     * Entry point of Duke. Simulates and main application loop.
+     * @param args
+     */
+    public static void main(String[] args) {
+        Ui duke = new Ui();
+        duke.start();
+        Scanner sc = new Scanner(System.in);
+
+        /* Initial setting of password */
+        while (!duke.checkAuth()) {
+            System.out.println("Please set a user password for auth.");
+            String p1 = sc.nextLine();
+            System.out.println("Please confirm password");
+            String p2 = sc.nextLine();
+            if (p1.trim().equals(p2.trim())) {
+                duke.setPw(p1.trim());
+            }
+        }
+
+        /* Main App Loop */
+        int x = 1;
+        //Loop until exit command given
+        while (x > 0) {
+            if (sc.hasNext()) {
+                x = duke.takeInputAndReturn(sc.nextLine());
+            } else {
+                duke.end();
+            }
+        }
+        if (x < 0) {
+            System.out.println("Fatal system error uncaught in main logic");
+        }
+    }
+
+    /**
+     * The main entry point for all JavaFX applications.
+     * The start method is called after the init method has returned,
+     * and after the system is ready for the application to begin running.
+     *
+     * <p>
+     * NOTE: This method is called on the JavaFX Application Thread.
+     * </p>
+     *
+     * @param primaryStage the primary stage for this application, onto which
+     *                     the application scene can be set.
+     *                     Applications may create other stages, if needed, but they will not be
+     *                     primary stages.
+     * @throws Exception if something goes wrong
```

Comment:
Can consider being replacing "if something goes wrong" in the javadoc to some thing more specific. I'm not sure what because I haven't done the java FX portion but if its meant to be generic then its fine!